src/goto-harness/memory_snapshot_harness_generator.cpp

```diff
@@ -378,3 +378,113 @@ void memory_snapshot_harness_generatort::generate(
 
   goto_functions.update();
 }
+
+memory_snapshot_harness_generatort::entry_goto_locationt::entry_goto_locationt(
+  const std::string &cmdl_option)
+{
+  std::vector<std::string> start;
+  split_string(cmdl_option, ':', start, true);
+
+  if(
+    start.empty() || start.front().empty() ||
+    (start.size() == 2 && start.back().empty()) || start.size() > 2)
+  {
+    throw invalid_command_line_argument_exceptiont(
+      "invalid initial location specification", "--initial-location");
+  }
+
+  function_name = start.front();
+
+  if(start.size() == 2)
+  {
+    location_number = optionalt<unsigned>(safe_string2unsigned(start.back()));
```

Comment:
this should be `string2optional_unsigned` instead, and you should check the result... if this function returns `nullopt` it means its argument wasn't a valid integer, which is something you don't want to silently ignore

(with `safe_string2unsigned` you're basically asserting that this conversion will always succeed which is not correct for user supplied strings)

Reply:
Done.